Musicca/app/src/main/java/com/example/musicca/activities/SongQueueActivity.java

```diff
@@ -0,0 +1,133 @@
+        btnBack = findViewById(R.id.btnBack);
+        btngotoPlaylist = findViewById(R.id.btngotoPlaylist);
+
+        songObjectId = getIntent().getStringExtra("songObjectid");
+        playlistObjectId = getIntent().getStringExtra("playlistobjectid");
+
+        albumUrl = getIntent().getStringExtra("albumiconurl");
+        Glide.with(this).load(albumUrl).into(ivSongAlbum);
+        tvTitle.setText(getIntent().getStringExtra("songtitle"));
+        tvArtist.setText(getIntent().getStringExtra("songartist"));
+
+        currentPlaylist = getCurrentPlaylist(playlistObjectId);
+        currentSong = getCurrentSong(songObjectId);
+
+        btnAddSong.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                addSong(currentSong);
+            }
+        });
+        btngotoPlaylist.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                gotoPlaylist();
+            }
+        });
+        btnBack.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                backToQueue();
+            }
+        });
+    }
+
+    private void gotoPlaylist() {
+        Intent i = new Intent(this, CurrentPlaylistActivity.class);
+        i.putExtra("playlistobjectid", playlistObjectId);
+        startActivity(i);
+    }
+
+    private void backToQueue() {
+        Intent i = new Intent(this, QueueActivity.class);
+        startActivity(i);
+    }
+
+    private Playlist getCurrentPlaylist(String playlistobjectid) {
+        final Playlist[] currentplaylist = new Playlist[1];
+        ParseQuery<Playlist> query = ParseQuery.getQuery(Playlist.class);
+        // First try to find from the cache and only then go to network
+        query.setCachePolicy(ParseQuery.CachePolicy.CACHE_ELSE_NETWORK); // or CACHE_ONLY
+        // Execute the query to find the object with ID
+        query.getInBackground(playlistobjectid, new GetCallback<Playlist>() {
+            @Override
+            public void done(Playlist playlist, com.parse.ParseException e) {
+                currentplaylist[0] = playlist;
+            }
+        });
+        return currentplaylist[0];
+    }
+    private Song getCurrentSong(String songobjectid) {
+        final Song[] currentsong = new Song[1];
+        ParseQuery<Song> query = ParseQuery.getQuery(Song.class);
+        // First try to find from the cache and only then go to network
+        query.setCachePolicy(ParseQuery.CachePolicy.CACHE_ELSE_NETWORK); // or CACHE_ONLY
+        // Execute the query to find the object with ID
+        query.getInBackground(songobjectid, new GetCallback<Song>() {
+            @Override
+            public void done(Song song, ParseException e) {
+                currentsong[0] = song;
+            }
+        });
+        return currentsong[0];
+    }
```

Comment:
It seems these method are basically the same with the ones in  Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java.
What are the differences between these files?
Can you extract these as helper methods to avoid code duplication?

Reply:
The SongQueueActivity displays song details from the song database, the SongPlaylistActivity displays song details added in a playlist and has the play song controls